packages/Operators/test/tstL2Projection.cpp

```diff
@@ -0,0 +1,186 @@
+#include <cmath>
+#include <cstdlib>
+#include <sstream>
+#include <algorithm>
+#include <cassert>
+
+#include "reference_implementation/DTK_ReferenceHexMesh.hpp"
+
+#include <DTK_FieldMultiVector.hpp>
+
+#include <DTK_L2ProjectionOperator.hpp>
+
+#include <DTK_BasicEntityPredicates.hpp>
+
+#include <Teuchos_UnitTestHarness.hpp>
+#include <Teuchos_DefaultComm.hpp>
+#include <Teuchos_RCP.hpp>
+#include <Teuchos_Array.hpp>
+#include <Teuchos_ParameterList.hpp>
+
+//---------------------------------------------------------------------------//
+// TEST EPSILON
+//---------------------------------------------------------------------------//
+
+const double epsilon = 1.0e-7;
+
+//---------------------------------------------------------------------------//
+// Tests
+//---------------------------------------------------------------------------//
+TEUCHOS_UNIT_TEST( L2ProjectionOperator, l2_projection )
+{
+    using namespace DataTransferKit;
+
+    // Get the communicator.
+    Teuchos::RCP<const Teuchos::Comm<int> > comm =
+	Teuchos::DefaultComm<int>::getComm();
+    DataTransferKit::LocalEntityPredicate local_pred( comm->getRank() );
+
+    // Set the global problem bounds.
+    double x_max = 3.1;
+    double y_max = 5.2;
+    double z_max = 8.3;
+    
+    // Create a source mesh and field.
+    int num_sx = 8;
+    int num_sy = 8;
+    int num_sz = 8;
+    double sx_width = x_max / num_sx;
+    double sy_width = y_max / num_sy;
+    double sz_width = z_max / num_sz;    
+    Teuchos::Array<double> sx( num_sx );
+    Teuchos::Array<double> sy( num_sy );
+    Teuchos::Array<double> sz( num_sz );
+    for ( int i = 0; i < num_sx; ++i ) sx[i] = i*sx_width;
+    for ( int i = 0; i < num_sy; ++i ) sy[i] = i*sy_width;
+    for ( int i = 0; i < num_sz; ++i ) sz[i] = i*sz_width;
+
+    DataTransferKit::UnitTest::ReferenceHexMesh source_mesh( comm, sx, sy, sz );
+    auto source_field = source_mesh.nodalField( 1 );
+    Teuchos::RCP<DataTransferKit::FieldMultiVector> source_vector =
+        Teuchos::rcp( new DataTransferKit::FieldMultiVector(comm, source_field) );
+
+    // Put some data on the source field.
+    double data_x = 9.3;
+    double data_y = 2.2;
+    double data_z = 1.33;
+    auto source_nodes = source_mesh.functionSpace()->entitySet()->entityIterator(
+        0, local_pred.getFunction() );
+    auto source_nodes_begin = source_nodes.begin();
+    auto source_nodes_end = source_nodes.end();
+    for ( source_nodes = source_nodes.begin();
+          source_nodes != source_nodes.end();
+          ++source_nodes )
+    {
+        int k = source_nodes->id() / (num_sx*num_sy);
+        int j = (source_nodes->id() - k*num_sx*num_sy) / num_sx;
+        int i = source_nodes->id() - j*num_sx - k*num_sx*num_sy;
+        TEST_EQUALITY( source_nodes->id(), i + j*num_sx + k*num_sx*num_sy );
+
+        double data = sx[i]*data_x + sy[j]*data_y + sz[k]*data_z + 1.0;
+        source_field->writeFieldData( source_nodes->id(), 0, data );
+    }
```

Comment:
So you build a field `f : (x, y, z) -> f(x, y, z) = data_x * x + data_y * y + data_z * z + 1`.
Be more explicit in the comments.